Formula/mydumper.rb

```diff
@@ -26,7 +26,7 @@ def install
     system "cmake", ".", *std_cmake_args,
            # Override location of mysql-client:
            "-DMYSQL_CONFIG_PREFER_PATH=#{Formula["mysql-client"].opt_bin}",
-           "-DMYSQL_LIBRARIES=#{Formula["mysql-client"].opt_lib}/libmysqlclient.dylib",
+           "-DMYSQL_LIBRARIES=#{Formula["mysql-client"].opt_lib}/#{shared_library("libmysqlclient")}",
```

Comment:
```suggestion
           "-DMYSQL_LIBRARIES=#{Formula["mysql-client"].opt_lib/shared_library("libmysqlclient")}",
```

Reply:
Updated. Also restructured a bit to avoid macOS-specific ZLIB logic.